flytekit/types/directory/__init__.py

```diff
@@ -26,3 +27,12 @@
     This is usually the SummaryWriter output in PyTorch or Keras callbacks which record the history readable by
     TensorBoard.
 """
+
+
+tfrecords_dir = typing.TypeVar("tfrecords_dir")
+TFRecordsDirectory = FlyteDirectory[tfrecords_dir]
```

Comment:
This format should be modified (see https://github.com/jerempy/flytekit/blob/bcab554eb69e31ca1844bf5c5a06a45b401e45f2/flytekit/types/file/__init__.py PR for reference). You can merge the latest master branch to fetch and merge the updates.